unified-runtime/source/loader/layers/sanitizer/sanitizer_common/sanitizer_utils.cpp

```diff
@@ -31,9 +31,12 @@ ur_usm_type_t GetUSMType(ur_context_handle_t Context, const void *MemPtr) {
 } // namespace
 
 ManagedQueue::ManagedQueue(ur_context_handle_t Context,
-                           ur_device_handle_t Device) {
+                           ur_device_handle_t Device, bool IsOutOfOrder) {
+  ur_queue_properties_t Prop{UR_STRUCTURE_TYPE_QUEUE_PROPERTIES, nullptr,
+                             UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE |
+                                 UR_QUEUE_FLAG_SUBMISSION_BATCHED};
```

Comment:
why `UR_QUEUE_FLAG_SUBMISSION_BATCHED`? This is just a hint, and the adapter is free to ignore it (and it will in many cases)

Reply:
We use out of order queue and flag UR_QUEUE_FLAG_SUBMISSION_BATCHED for better performance in some cases. And  UR_QUEUE_FLAG_SUBMISSION_BATCHED is mainly for l0 gpu device.

Reply:
Can you elaborate on scenarios where you saw a performance benefit?

We will soon stop supporting batched submission at all on L0. In the new v2 adapter this mode is not supported and this hint is ignored.

Reply:
Thanks for the notification. If new v2 adapter doesn't support batch mode, is there another way to disable l0 using immediate command list?

Reply:
Immediate Command Lists are going to be the only supported submission mode going forward.

Reply:
Ok, got it. I will revert out of queue related changes and have a try for command buffers.

Reply:
@pbalcer, out of order queue changes have been reverted. Please continue to review. Many thanks.